sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusAsyncConsumer.java

```diff
@@ -75,7 +108,9 @@ Mono<Void> updateDisposition(String lockToken, DispositionStatus dispositionStat
     @Override
     public void close() {
         if (!isDisposed.getAndSet(true)) {
-            linkProcessor.dispose();
+            if (!isV2) {
```

Comment:
thanks! Looking at this close() method again, we can even remove the `isDisposed` member because it’s not used anywhere else and v1 `linkProcessor.dispose()` already has logic in place to NOP if called more than once, so we can have -

```java
if (isV2) {
   return;
}
linkProcessor.dispose();
```